### PR TITLE
Add crud data store creator

### LINKED ISSUE #33435: Get/list all product attributes via products data store API
### Description

Add the code to be able to get the product attributes through the products data store.

_For now, this should be marked as experimental/internal as we iterate on the API so that it is clear that this is incomplete and subject to change_.

#### _Get_
**HTTP Request**
> **GET** `/wp-json/wc/v3/products/attributes/<id>`

**Reference**
> https://woocommerce.github.io/woocommerce-rest-api-docs/#retrieve-a-product-attribute

#### _List all_
**HTTP Request**
> **GET** `/wp-json/wc/v3/products/attributes`

**Properties**
| Parameter | Type | Description |
| --------- | -------- | -------- |
| `context` | string | Scope under which the request is made; determines fields present in response. Options: `view` and `edit`. Default is `view`. |

**Reference**
> https://woocommerce.github.io/woocommerce-rest-api-docs/#list-all-product-attributes

### Acceptance criteria
- [ ] Data store updates
- [ ] Unit tests
- [ ] Documentation with examples

### PR DESCRIPTION
**Pull request by joshuatf**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

Adds a crud data store creator to make implementing of data stores easier.

Also adds the get/list product attributes methods.

Closes #33435  .

### How to test the changes in this Pull Request:

1. Create some product attributes under Products->Attributes
2. Open your console
3. Enter the following: `wp.data.select('wc/admin/products/attributes').getProductAttributes( {} );`
4. You may have to rerun that command after fetching has resolved
5. Note that attributes are shown
6. Using an ID of one of your attributes, run the following replacing `3` with your attribute ID: `wp.data.select('wc/admin/products/attributes').getProductAttribute( 3 );`
7. You may have to rerun that command after fetching has resolved if you did not previously fetch attributes
8. Note that the attribute is shown
9. Check that the README info looks sound

### Other information:

-   [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [ ] Have you successfully run tests with your changes locally?
-   [ ] Have you created a changelog file for each project being changed, ie `pnpm changelog add --filter=<project>`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Comment by botwoo** [comment]
:bar_chart: Test reports for this pull request have been published and are accessible through the following links:

- [REST API Test Report](https://woocommerce.github.io/woocommerce-test-reports/pr/33546/api/)
- E2E Test Report (TBC)

Latest commit referenced in the reports: `Remove unused vars` 8fa46bf874b96ba0889363f2e939d1b22f36ea4c
This comment will automatically be updated with the latest referenced commit when you push new changes to this pull request.

---

_Visit the [WooCommerce Test Reports](https://woocommerce.github.io/woocommerce-test-reports) homepage to view all published reports. See the [FAQs](https://github.com/woocommerce/woocommerce-test-reports/wiki/FAQs) page if you're having problems accessing them._

File: packages/js/data/src/crud/test/reducer.ts [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+		expect( state.data[ 2 ].title ).toEqual( initialState.data[ 2 ].title );
+		expect( state.data[ 2 ].status ).toEqual( update.status );
+	} );
+
+	it( 'should handle GET_ITEMS_SUCCESS', () => {
+		const items: Item[] = [
+			{ id: 1, name: 'Yum!' },
+			{ id: 2, name: 'Dynamite!' },
+		];
+		const totalCount = 45;
```
Comment (octaedro) [inline packages/js/data/src/crud/test/reducer.ts]: I think that this `const` is not being used.

File: packages/js/data/src/crud/test/reducer.ts [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+					description: 'dynamite',
+				},
+			},
+		};
+
+		const items: Item[] = [
+			{ id: 1, name: 'Yum! Updated' },
+			{ id: 2, name: 'Dynamite!' },
+		];
+		const totalCount = 45;
```
Comment (octaedro) [inline packages/js/data/src/crud/test/reducer.ts]: Same with this `const`

**Review by octaedro — DISMISSED** [review]
You did a great job here @joshuatf! The code looks good and is testing well on my end. I left a couple of tiny comments. The rest looks great to me.

**Review by AnnaMag — DISMISSED** [review]
LGTM

**Comment by joshuatf** [comment]
Thanks, @octaedro and @AnnaMag!  Removed those unused vars in https://github.com/woocommerce/woocommerce/pull/33546/commits/8fa46bf874b96ba0889363f2e939d1b22f36ea4c.

**Review by AnnaMag — APPROVED** [review]
🚢

**Review by octaedro — APPROVED** [review]
Thank you for removing the vars. LGTM :shipit:

**Pull request merged**